episodes/instance-segmentation-and-measurements.md

```diff
@@ -0,0 +1,596 @@
+Looking at the image above, there are no longer any incorrectly joined
+nuclei. The absolute number of nuclei found hasn't changed much as the
+erosion process has removed some partial nuclei around the edges of the
+image.
+
+Performing any size or shape analysis on these nuclei will be flawed, as
+they are heavily eroded. We can largely undo much of the erosion by using
+the scikit-images [expand labels](
+https://scikit-image.org/docs/stable/api/skimage.segmentation.html#skimage.segmentation.expand_labels)
+function.
+
+```python
+from skimage.segmentation import expand_labels
+
+# Remove the eroded instance segmentation from the viewer
+viewer.layers.remove('instance_seg')
+
+# Expand the labels using the same radius we used when eroding them (10)
+instance_seg = expand_labels(instance_seg, 10)
+
+# Put the new instance segmentation back in the viewer
+viewer.add_labels(instance_seg)
+```
+![](fig/instance_segmentation_expanded.png){
+alt="Expanded instance segmentation on the eroded segmentation mask"}
+There are now 19 apparently correctly labelled nuclei that appear to be
+the same shape as in the original mask image.
+
+:::::::::::::::::::::::::challenge
+
+### Is the erosion completely reversible?
+In order to create a correct instance segmentation we have performed a
+mask erosion followed by a label expansion. This is a common image
+operation often used to remove back ground noise, known as as `opening`,
+or an erosion followed by a dilation. In addition to helping us separate
+instances it will have the effect of removing objects smaller than the
+erosion mask, in this case a sphere with radius 10 pixels.
+If we compare the eroded and expanded image with the original mask,
+what will we see?
+
+:::::::::::::::::::::::::solution
+
+![](fig/instance_segmentation_vs_semantic_segmentation.png){
+alt="A comparison between the expanded instance segmentation and the
+original semantic segmentation showing some mismatch between the borders."}
+Looking at the above image we can see some small mismatches around the
+edges of most of the nuclei. Will the effect of this on the accuracy of
+our results be significant?
+
+:::::::::::::::::::::::::
+:::::::::::::::::::::::::
+## Removing Border Cells
+Now we return to the second problem with our initial instance segmentation,
+the presence of partial nuclei around the image borders.
+We can remove these from our analysis using scikit-image's
+[clear border](
+https://scikit-image.org/docs/stable/api/skimage.segmentation.html#skimage.segmentation.clear_border)
+function.
+
+```python
+
+from skimage.segmentation import clear_border
+
+viewer.layers.remove('instance_seg')
+
+instance_seg = clear_border(instance_seg)
+
+viewer.add_labels(instance_seg)
+```
+
+![](fig/instance_segmentation_clear_border.png){
+alt="The instance segmentation with any nuclei crossing the image boundary
+removed"}
+We now have an image with 11 clearly labelled nuclei. Let's check the
```

Comment:
Why is the funny, bright shaped one in the upper left - why has clear borders removed this one? It looks well within the border on this slide, does it touch another border somewhere?

Reply:
```suggestion
We now have an image with 11 clearly labelled nuclei. 
You may notice that the smaller nucleus (dark orange) near the top left
of the image has been removed even though we can't see where it touches the
image border. Remember that this is a 3D image and clear border removes
nuclei touching any border. This nucleus has been removed because it touches
the top or bottom (z axis) of the image.
Let's check the
```